tests/rptest/tests/tx_abort_index_test.py

```diff
@@ -0,0 +1,136 @@
+            'transactional.id': '1',
+        })
+        p.init_transactions()
+        for _ in range(0, 5):
+            p.begin_transaction()
+            p.produce(topic,
+                      key="key1".encode('utf-8'),
+                      value=PAYLOAD_1KB.encode('utf-8'),
+                      callback=on_delivery)
+            p.flush()
+            p.abort_transaction()
+
+    def fill_segment(self, topic):
+        p = Producer({
+            "bootstrap.servers": self.redpanda.brokers(),
+            "enable.idempotence": True,
+            "retries": 5
+        })
+        for _ in range(0, 4 * 1024):
+            p.produce(topic,
+                      key="key1".encode('utf-8'),
+                      value=PAYLOAD_1KB.encode('utf-8'),
+                      callback=on_delivery)
+        p.flush()
+
+    def find_indexes(self, topic):
+        idxes = dict()
+        for node in self.redpanda.nodes:
+            idxes[node.account.hostname] = []
+            cmd = f"find {RedpandaService.DATA_DIR}"
+            out_iter = node.account.ssh_capture(cmd)
+            for line in out_iter:
+                m = re.match(
+                    f"{RedpandaService.DATA_DIR}/kafka/{topic}/\\d+_\\d+/(abort.idx.\\d+.\\d+)",
+                    line)
+                if m:
+                    idxes[node.account.hostname].append(m.group(1))
+        return idxes
+
+    def find_segments(self, topic):
+        segments = dict()
+        for node in self.redpanda.nodes:
+            segments[node.account.hostname] = []
+            cmd = f"find {RedpandaService.DATA_DIR}"
+            out_iter = node.account.ssh_capture(cmd)
+            for line in out_iter:
+                m = re.match(
+                    f"{RedpandaService.DATA_DIR}/kafka/{topic}/\\d+_\\d+/(.+).log",
+                    line)
+                if m:
+                    self.logger.info(f"{node.account.hostname} {line}")
+                    segments[node.account.hostname].append(m.group(1))
+        return segments
+
+    @cluster(num_nodes=3, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_index_removal(self):
+        self.fill_idx(TOPIC)
+        segments = self.find_segments(TOPIC)
+        for node in self.redpanda.nodes:
+            assert len(segments[node.account.hostname]) == 1
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+        idx = self.find_indexes(TOPIC)
+        for node in self.redpanda.nodes:
+            assert len(idx[node.account.hostname]) > 0
+        self.fill_segment(TOPIC)
+
+        def segments_gone():
+            current = self.find_segments(TOPIC)
+            for node in segments.keys():
+                if node not in current:
+                    continue
+                if segments[node][0] in current[node]:
+                    return False
+            return True
+
+        wait_until(segments_gone, timeout_sec=60, backoff_sec=1)
+
+        self.redpanda.restart_nodes(self.redpanda.nodes)
```

Comment:
What is triggering the cleanup (deletion) to happen? Is it this restart?

Reply:
The cleanup happens when redpanda creates a snapshot, several events cause a snapshot creation, one of them is the startup so the restart indirectly causes the deletion.